openedx/features/course_experience/views/course_sock.py

```diff
@@ -0,0 +1,53 @@
+"""
+Fragment for rendering the course's verification sock and associated toggle button.
+"""
+from datetime import datetime
+
+from django.conf import settings
+from django.template.loader import render_to_string
+from opaque_keys.edx.keys import CourseKey
+from web_fragments.fragment import Fragment
+
+from student.models import CourseEnrollment
+from course_modes.models import CourseMode
+from courseware.date_summary import VerifiedUpgradeDeadlineDate
+from courseware.courses import get_course_with_access
+from courseware.views.views import get_course_prices
+from openedx.core.djangoapps.plugin_api.views import EdxFragmentView
+
+
+class CourseSockFragmentView(EdxFragmentView):
+    """
+    A fragment to allow users to verify the course from the home page
```

Comment:
Nit: I'd make this more generic, because the sock won't always be about upgrading.